File: loops.py
# Loops o Bucles
"""

for iterador in datos:
    sentencias

while condicion:
    sentencias


range(start, stop, step) start = 0 stop = ? step = 1

variable += 1
variable = variable + 1

"""

import random
import math
nombres = ['Hugo', 'Paco', 'Luis']

# for with range
for i in range(0, 11, 2):
    print(i)

# for in with range
for indice in range(0, len(nombres)):
    print(nombres[indice])

# for in
for nombre in nombres:
    print(nombre)

i = 0
while(i < len(nombres)):
    print(nombres[i])
    i+=1


uppercase_names = [ name.upper() for name in nombres]
print(uppercase_names)

numbers_to_hundred = [ math.floor(random.random() * num) for num in range(1, 11)]
print(numbers_to_hundred)

numbers = []
for x in range(10):
    numbers.append(math.floor(random.random() * 10) + 1)

print(numbers)

File: loops.js
// Loops o Bucles
/*  

for(iterador; condicion; incremento){
    sencencias
}

for(indice in coleccion){
    sentencias
}

for(valor of coleccion){
    sentencias
}

while(condicion){
    sentencias
}

do {
    sentencias
} while (condicion);

*/

for (let i = 1; i <= 10; i++) {
    console.log(i);
}

let nombres = ['Hugo', 'Paco', 'Luis'];

for (let i = 0; i < nombres.length; i++) {
    console.log(nombres[i]);
}

for (let indice in nombres) {
    console.log(nombres[indice]);
}

for (let valor of nombres) {
    console.log(valor);
}

let j = 1;
while (j <= 10) {
    console.log(j);
    j++;
}

let k = 0;
while (k < nombres.length) {
    console.log(nombres[k]);
}

let m = 0;
do {
    console.log(nombres[m]);
    m++;
} while (m == 10)